evmrpc/worker_pool_metrics.go

```diff
@@ -0,0 +1,874 @@
+package evmrpc
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	gometrics "github.com/armon/go-metrics"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+)
+
+// Environment variable to enable debug metrics printing to stdout
+// Set EVM_DEBUG_METRICS=true to enable periodic metrics printing
+const EVMDebugMetricsEnvVar = "EVM_DEBUG_METRICS"
+
+// IsDebugMetricsEnabled checks if debug metrics printing is enabled via environment variable
+func IsDebugMetricsEnabled() bool {
+	val := os.Getenv(EVMDebugMetricsEnvVar)
+	return strings.ToLower(val) == "true" || val == "1"
+}
+
+// Error type constants for categorization
+const (
+	ErrTypeRangeTooLarge = "range_too_large"
+	ErrTypeTimeout       = "timeout"
+	ErrTypeRateLimited   = "rate_limited"
+	ErrTypeBackpressure  = "backpressure"
+	ErrTypeIOSaturated   = "io_saturated"
+	ErrTypeBlockNotFound = "block_not_found"
+	ErrTypeQueueFull     = "queue_full"
+	ErrTypeOther         = "other"
+)
+
+// WorkerPoolMetrics tracks worker pool performance metrics
+type WorkerPoolMetrics struct {
```

Comment:
Nonblocker: Sad that this was not implemented using otel. Is it too late to do that instead of adding to the well out of date telemetry and go-metrics stuff considering this repo is already set up with otel?